packages/mui-system/src/createStyled.js

```diff
@@ -163,19 +178,117 @@ export default function createStyled(input = {}) {
             // On the server Emotion doesn't use React.forwardRef for creating components, so the created
             // component stays as a function. This condition makes sure that we do not interpolate functions
             // which are basically components used as a selectors.
-            return typeof stylesArg === 'function' && stylesArg.__emotion_real !== stylesArg
-              ? (props) => {
-                  return stylesArg({
-                    ...props,
-                    theme: resolveTheme({ ...props, defaultTheme, themeId }),
-                  });
+            if (typeof stylesArg === 'function' && stylesArg.__emotion_real !== stylesArg) {
```

Comment:
Adding the variants styled in case arrays are used for defining multiple styles.